File: prelude/toolchains/android/src/com/facebook/buck/jvm/java/JavacLanguageLevelOptions.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under both the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree and the Apache
 * License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 * of this source tree.
 */

package com.facebook.buck.jvm.java;

import com.facebook.buck.core.util.immutables.BuckStyleValueWithBuilder;
import com.facebook.buck.jvm.java.version.JavaVersion;
import org.immutables.value.Value;

@BuckStyleValueWithBuilder
public abstract class JavacLanguageLevelOptions {

  public static final JavacLanguageLevelOptions DEFAULT =
      JavacLanguageLevelOptions.builder().build();

  // Default combined source and target level.
  public static final String TARGETED_JAVA_VERSION = "7";

  @Value.Default
  protected String getSourceLevel() {
    return TARGETED_JAVA_VERSION;
  }

  @Value.Default
  protected String getTargetLevel() {
    return TARGETED_JAVA_VERSION;
  }

  @Value.Derived
  public JavaVersion getSourceLevelValue() {
    return JavaVersion.toJavaLanguageVersion(getSourceLevel());
  }

  @Value.Derived
  public JavaVersion getTargetLevelValue() {
    return JavaVersion.toJavaLanguageVersion(getTargetLevel());
  }

  public static Builder builder() {
    return new Builder();
  }

  public static class Builder extends ImmutableJavacLanguageLevelOptions.Builder {}
}

File: prelude/toolchains/android/src/com/facebook/buck/jvm/java/JavacLanguageLevelOptions.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is dual-licensed under either the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree or the Apache
 * License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 * of this source tree. You may select, at your option, one of the
 * above-listed licenses.
 */

package com.facebook.buck.jvm.java

import com.facebook.buck.jvm.java.version.JavaVersion

data class JavacLanguageLevelOptions(val sourceLevel: String, val targetLevel: String) {
  val sourceLevelValue: JavaVersion
    get() = JavaVersion.toJavaLanguageVersion(sourceLevel)

  val targetLevelValue: JavaVersion
    get() = JavaVersion.toJavaLanguageVersion(targetLevel)

  companion object {
    // Default combined source and target level.
    const val TARGETED_JAVA_VERSION: String = "7"

    @JvmField
    val DEFAULT: JavacLanguageLevelOptions =
        JavacLanguageLevelOptions(TARGETED_JAVA_VERSION, TARGETED_JAVA_VERSION)
  }
}
